rosetta-source/src/main/rosetta/event-common-func.rosetta

```diff
@@ -1661,6 +1661,87 @@ func Qualify_Shaping: <"The qualification of a shaping event from the fact that
 		and packageRef count = openTradeNoExecutionDetails count
 		and packageRef distinct count = 1
 
+
+func Create_PartialDeliveryPrimitiveInstruction: <"Creates the primitive instructions for a substitution of collateral by replacing the assetpayout of the trade.">
+	inputs:
+		//ICMA-P2 Instructions
+		tradeState TradeState (1..1) <"The original trade to be closed.">
+		effectiveDate AdjustableOrRelativeDate (1..1) <"The date to close and open a new position.">
+		terminationDate AdjustableOrRelativeDate (1..1) <"The new termination date.">
+		deliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+		undeliveredPriceQuantity PriceQuantity (1..*) <"The price and quantity of the remaining trade. ">
+	output: 
```

Comment:
Do we need undeliveredPriceQuantity? If this is a partial delivery then you would just specify what has been delivered and the remainder would be calculated surely? Just wondering if we need both amounts to be specified.